Feature/platformio

andre-stefanov opened the pull request:
This PR introduces usage of [PlatformIO](https://platformio.org/). Following changes were made:
- most .ino files were renamed to .hpp (only OpenAstroTracker.ino kept its name to keep support for Arduino IDE)
- OpenAstroTracker.ino includes previous .ino files in alphabetical order to mimic previos behavior of Arduino IDE
- added `#pragma once` to former .ino files (now .hpp)
- in **c722_menuPOI.hpp** the POI "M42 Orion Nbula" had to be commented out for now because it uses an invalid byte (unit8) value -5 and causes compile error. There is a fix by @ClutchplateDude on its way.
- added UNO, MEGA2560 and ESP8266 as supported platforms
- added `void processSerialData();` declaration on top of **f_seial.hpp** because it was calling this function before it was defined
- added `#include "f_serial.hpp"` to **c_buttons.hpp** because it depends on serial without including it. This dependency was somehow fixed by Arduino IDE before.
- removed VMicro configurations because they would be broken due to the file renamings and it is easier to generate them anew instead of me touching them manually without a possibility to test them. Feel free to generate them and push to this PR.

ClutchplateDude commented on Software/Arduino code/OpenAstroTracker/OpenAstroTracker.sln:
```diff
@@ -1,24 +0,0 @@
-﻿
```
This file should not be deleted, it is required for Visual Studio

andre-stefanov replied:
Yes but keeping it in old state would break the visual studio project (file names changed). I removed these files so you can import the project again in VS and this way generate correct project files.

ClutchplateDude commented on Software/Arduino code/OpenAstroTracker/OpenAstroTracker.vcxproj:
```diff
@@ -1,202 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
```
This file should not be deleted, it is required for Visual Studio

andre-stefanov replied:
See other comment.

ClutchplateDude commented on Software/Arduino code/OpenAstroTracker/OpenAstroTracker.vcxproj.filters:
```diff
@@ -1,105 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
```
This file should not be deleted, it is required for Visual Studio

andre-stefanov replied:
See other comment.

jwellman80 commented:
I have no issue with this PR.  The VSMicro stuff can be easily regenerated.  I need to play with PlatformIO portion a little more though.

andre-stefanov commented:
I have improved the configuration. Added Esp32 and fixed the definitions like ESP32 not being recognized properly. The only thing i am not completelly sure of was a required adaption to the TimerInterrupt.cpp, which was lacking a `#if !defined(ESP32)` and thus included the `TimerInterrupt.h`. I have extended the macro but since these files are actually a library, i dont know if we really should touch them.
